### PR TITLE
updated routes / controller, entity, service for user Module

### PR DESCRIPTION
**Pull request by mathildegrimal**
#### Before submitting, make sure:
- [x] You have rebased your branch
- [x] You describe your changes
- [x] You describe How to test
- [x] I work on the same file as one of my colleague which one?

#### Describe below your changes:
Changes in the user module (controller's routes with try and catch, entity updated, service updated with exceptions managing)

##### Describe below How to test:
no test by now

### CODE REVIEW

**Comment by mathildegrimal** [comment]
[En tant que développeur je souhaite pouvoir accéder à un CRUD User](https://trello.com/c/t06I0EtJ/38-en-tant-que-d%C3%A9veloppeur-je-souhaite-pouvoir-acc%C3%A9der-%C3%A0-un-crud-user)

File: src/users/exceptions/email-already-used.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+
+export class EmailAlreadyUsed extends HttpException {
+  constructor(email: string) {
+    super(`"${email}" is already used`, HttpStatus.CONFLICT);
```
Comment (Antonin-dev) [inline src/users/exceptions/email-already-used.ts:5]: super(`${email} is already used`, HttpStatus.CONFLICT);

Comment (Antonin-dev) [reply src/users/exceptions/email-already-used.ts:5]: can you remote " " around email

File: src/users/exceptions/user-not-found.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+
+export class UserNotFound extends HttpException {
+  constructor(id: string) {
+    super(`User with id "${id}" doesn't exist`, HttpStatus.NOT_FOUND);
```
Comment (Antonin-dev) [inline src/users/exceptions/user-not-found.ts:5]: can you remove double quote

File: src/users/users.controller.ts [hunk 3]
```diff
@@ -8,11 +8,20 @@ import {
   UsePipes,
   ValidationPipe,
   ParseIntPipe,
```
Comment (Antonin-dev) [inline src/users/users.controller.ts:10]: Can you remove unused variable?

File: src/users/exceptions/user-not-found.ts (hunk 2 above)
Comment (mathildegrimal) [reply src/users/exceptions/user-not-found.ts:5]: Yes I can it was just for a better visualisation

File: src/users/exceptions/email-already-used.ts (hunk 1 above)
Comment (mathildegrimal) [reply src/users/exceptions/email-already-used.ts:5]: yes I can

Comment (mathildegrimal) [reply src/users/exceptions/email-already-used.ts]: Done

File: src/users/exceptions/user-not-found.ts (hunk 2 above)
Comment (mathildegrimal) [reply src/users/exceptions/user-not-found.ts]: done

File: src/users/users.controller.ts (hunk 3 above)
Comment (mathildegrimal) [reply src/users/users.controller.ts:10]: yes

**Comment by mathildegrimal** [comment]
@Antonin-dev I made the changes and committed, can you check ?

**Review by Antonin-dev — APPROVED** [review]

**Comment by mathildegrimal** [comment]
Merge user-crud into développement

**Pull request merged**